web/custom-plugins/forecast-of-response/src/components/AccessControl/AccessControlList.tsx

```diff
@@ -0,0 +1,215 @@
+    if (props.even) return "#F7F7F7"
+    return "inherit"
+  }}
+`
+
+const StyledOption = styled.option`
+  font-weight: 700;
+`
+const StyledSelect = styled.select`
+  font-weight: 700;
+  width: 80px;
+  padding: 3px;
+  border-radius: 3px;
+  background-color: transparent
+`
+
+
+const ACLSelect = ({ value, handleChange }: any): JSX.Element => {
+  return (
+    <div style={{ width: "150px", padding: "10px" }}>
+      <StyledSelect value={value} onChange={(event: Event) => handleChange(event.target.value)}>
+        {Object.values(ACLEnum).map((accessLevel: string) =>
+          <StyledOption value={accessLevel} key={accessLevel}>{accessLevel}</StyledOption>)}
+      </StyledSelect>
+    </div>
+  )
+}
+
+const ACLOwnerPanel = ({ acl, handleChange }: any): JSX.Element => {
+  return <>
+    <CenteredRow width={"230px"}>
+      Owner:
+      <Input
+        style={{ width: "150px", marginLeft: "5px" }}
+        placeholder={acl.owner}
+        onChange={(event): Event => handleChange({ owner: event.target.value })}/>
+      <Icons name="edit_text" size={24} style={{ color: 'teal' }}/>
+    </CenteredRow>
+    <CenteredRow width={"160px"}>
+      Others:
+      <ACLSelect value={acl.others} handleChange={(newValue: string) => handleChange({ others: newValue })}/>
+    </CenteredRow>
+  </>
+}
+
+interface StringMap {
+  [key: string]: string
+}
+
+interface URPanelProps {
+  entities: StringMap
+  handleChange: Function
+  aclKey: string
+
+}
+
+const ACLUserRolesPanel = ({ entities, handleChange, aclKey }: URPanelProps): JSX.Element => {
+  const [newRole, setNewRole] = useState<string>("")
+  return <>
+    <CenteredRow>
+      <Input
+        style={{ width: "170px" }}
+        placeholder={"Add new role"}
+        onChange={(e): Event => setNewRole(e.target.value)}/>
+      <Button onClick={() => handleChange({ [aclKey]: { ...entities, [newRole]: "NONE" } })}>Add +</Button>
+    </CenteredRow>
+    <ListRow>
+      <div>Role</div>
+      <div>Access Level</div>
+      <div></div>
+    </ListRow>
+    <TableWrapper>
+      {Object.entries(entities).map(([entity, access], index): any => {
+        const roleHandleChange = (value: string) => {
+          entities[entity] = value
+          handleChange({ [aclKey]: entities })
+        }
+        return (
+          <GridContainer key={entity} even={index % 2 == 0}>
+            <div>{entity}</div>
+            <ACLSelect value={access} handleChange={roleHandleChange}/>
+              <Button variant="outlined" color="danger" onClick={() => {
+                  delete entities[entity]
+                  handleChange({[aclKey]: entities})
+              }}>
+                Remove
+              </Button>
+          </GridContainer>
+        )
+      })
+      }
+    </TableWrapper>
+  </>
+}
+
+export default (documentId: any): JSX.Element => {
+  const [activeTab, setActiveTab] = useState<number>(0)
+  const [documentACL, setDocumentACL] = useState<any|null>({
```

Comment:
we could perhaps define a type for the ACL document (to avoid use of any) 
(would probably be enough with something like: 
{
owner: str
roles: any
users: any
others: str
}

)

Reply:
fixed!